src/common/store/synchronization.slice.ts

```diff
@@ -70,18 +83,20 @@ export const getSynchronizationState = createSelector(
 
 export const getPublicKey = createSelector(getSynchronizationState, (state) => state.publicKey);
 export const getPrivateKey = createSelector(getSynchronizationState, (state) => state.privateKey);
-export const getHandshaken = createSelector(getSynchronizationState, (state) => state.isHandshaken);
+export const getHandshaken = (target: SynchronizationTarget) =>
+  createSelector(getSynchronizationState, (state) => state.isHandshaken[target]);
 
-export const getTargetPublicKey = createSelector(
-  getSynchronizationState,
-  (state) => state.targetPublicKey
-);
+export const getTargetPublicKey = (target: SynchronizationTarget) =>
+  createSelector(getSynchronizationState, (state) => state.targetPublicKey[target]);
 
 export const postHandshake = createAction(`${sliceName}/postHandshake`);
 
-export function* handshakeSaga(ipc: ReduxIPC) {
+export function* handshakeSaga(
+  ipcs: Partial<Record<SynchronizationTarget, ReduxIPC>>,
+  self: SynchronizationTarget
+) {
   yield all([
-    ipcWorker(ipc),
+    ...Object.keys(ipcs).map((target) => ipcWorker(ipcs, target as SynchronizationTarget, self)),
```

Comment:
What´s the reason for the type cast here? Is TypeScript not able to infer the type?

Reply:
No, it just assumes it is a string